python/arcticdb/arctic.py

```diff
@@ -151,6 +154,9 @@ def __init__(self, uri: str, encoding_version: EncodingVersion = EncodingVersion
         self._uri = uri
         self._open_libraries = dict()
 
+        if check_storage_access:
```

Comment:
I think it would be good to have a `check_storage_accessibility` method on the top-level `Arctic` API itself, that gives back to the user an enum describing the result of the check and any useful messages.

Reply:
I don't think there's much point in calling this after the constructor. The user could call a real read/write method at that point which more conclusively verifies the access permissions.

The call sequence is quite simple and easy to figure out (open source code).